### ISSUE TITLE
Default route not set

### ISSUE THREAD

**Issue by ertimas** [clemenko/rke_airgap_install#23]
Hi Andy,

I am running into a missing default ip route, which the k3s team solves via a dummy in this issue: https://github.com/k3s-io/k3s/issues/1144.
![image](https://github.com/user-attachments/assets/5fc2d5ad-2812-4cfd-8cd5-8fc250ccaea7)

It seems like you would have run into that?  Did you have this configured prior to the script, or did I miss something?

Thanks for your time!

**Comment by clemenko** [comment]
I have not seen that before. Let's look at the basics. 
What OS ? `cat /etc/os-release`
What nics ? `ip a`
did you modify the `config.yaml`?

**Comment by ertimas** [comment]
Hi Andy,
Thank you for getting back to me.
### OS
Oracle Linux 8.10 server

### NICS
![image](https://github.com/user-attachments/assets/6dcf2652-e96f-46e1-9a3f-5eda4cc7dfb3)

### Config

I didn't modify `config.yaml`, though I also didn't have the file in `/etc/rancher/rke2/`. So I symlinked one to the existing `/etc/rancher/rke2/rke2.yaml`.  Which got through that issue. Though I'm left with a new one....

![image](https://github.com/user-attachments/assets/6af68c96-5161-466c-8db6-ce30320e587f)

`hauler.repo` is in `fileserver/`.  
![image](https://github.com/user-attachments/assets/89a76b5f-f2c2-43b6-ae89-937f66ec8592)


Here's the output from netstat
![image](https://github.com/user-attachments/assets/7e8e33b4-fcd3-4cdd-a738-a12ff625ee03)


Any thought on how to troubleshoot fileserver not seeing `hauler.repo` would be much appreciated.  Note: my hauler directory is has global read/write permissions.

**Comment by clemenko** [comment]
the two issues may be related. 
A. routes
B. hauler. 

Can you try a clean install of Oracle 8 and see if any routes are there? `/proc/net/route` specifically?

**Comment by ertimas** [comment]
Here's `/proc/net/route`.  Note I've got a second interface attached so that I can get the VM setup before "airgapping" it

![image](https://github.com/user-attachments/assets/abc4b13b-da25-4e13-af4a-d8b8f21c76e4)

**Comment by clemenko** [comment]
I wonder is the error was an isolated issue. Can you try and install it again?

Wait, are you disabling the NIC when installing the stack? Kubernetes needs an NIC at all times.

**Comment by ertimas** [comment]
Last time I left both NICs on during installation.  I've tried it a couple times with each/both NICs, no ip given to `hauler_all_the_things.sh`, still the same issue

**Comment by clemenko** [comment]
there is no IP needed for the control function. can you run the control command with `bash -x ./hauler_all_the_things.sh control` ?

**Comment by ertimas** [comment]
I found that if I didn't provide an IP then it might look at the second NIC, and [it looks like this made](https://github.com/clemenko/rke_airgap_install/blob/main/hauler_all_the_things.sh#L36) it happen.

Here is the result of `bash -x ./hauler_all_the_things.sh control`

![image](https://github.com/user-attachments/assets/908fcbe4-c641-4b2d-b3d1-c034b97af347)

Here's the log where `createrepo` was run
![image](https://github.com/user-attachments/assets/17ec13c6-6638-42e6-9a83-ab5f9ec21110)

**Comment by clemenko** [comment]
Can you see if the hauler.repo file is there : `curl -sfL http://10.237.13.41:8080/hauler.repo`?

**Comment by ertimas** [comment]
I can see the file, but it's inaccessible if running the hauler server, see the yellow box.  Note: if I run `python3 http.server` I can access the file from another box, which leads me to believe basic networking and file permissions are fine.

![Screenshot 2024-09-17 at 09 24 12](https://github.com/user-attachments/assets/979ac0b1-187b-4414-b451-c661693bba0a)

**Comment by clemenko** [comment]
I wonder is Hauler is not able to server anything on a node with 2 nics. 
What does `ss -tln` show?

**Comment by ertimas** [comment]
It's shows the same thing as `netstat -lnt` essentially.


Manually running `hauler store serve fileserver <my store directory>` did work.  It took about a minute for it to come up.  Here are the debug logs in case they're helpful.  Marking this as closed

![image](https://github.com/user-attachments/assets/904df590-17d5-46f8-9829-c242b5491455)

**Comment by clemenko** [comment]
huh. I see you closed the issue. Is it working now?

**Comment by ertimas** [comment]
TLDR; I couldn't get your script to work.

There were two issues.

1. The fileserver service looks like it's fine, but after several minutes fails to be `curl`able.  So, I ran `hauler store serve -l debug fileserver <my hauler directory>` and it worked.  The command showed the fileserver taking >1minute for the to come up.  See the screenshot I posted....
2. The registry fails to load.  This is due to the a `longhorn-csi` component in the hauler store `.zst` being corrupted during tar/untar.  Removing the `csi` components from `airgap_hauler.yaml` fixed it.
